unblob/cli.py

```diff
@@ -453,35 +464,36 @@ def print_report(reports: ProcessResult):
             valid_size += size
         total_size += size
 
-    if total_size == 0:
-        return
+    console = Console()
 
-    summary = Panel(
-        f"""Extracted files: [#00FFC8]{total_files}[/#00FFC8]
+    if total_size:
+        summary = Panel(
+            f"""Extracted files: [#00FFC8]{total_files}[/#00FFC8]
 Extracted directories: [#00FFC8]{total_dirs}[/#00FFC8]
 Extracted links: [#00FFC8]{total_links}[/#00FFC8]
 Extraction directory size: [#00FFC8]{human_size(extracted_size)}[/#00FFC8]
 Chunks identification ratio: [#00FFC8]{(valid_size/total_size) * 100:0.2f}%[/#00FFC8]""",
-        subtitle="Summary",
-        title=f"unblob ({get_version()})",
-    )
+            subtitle="Summary",
+            title=f"unblob ({get_version()})",
+        )
 
-    console = Console()
-    console.print(summary)
+        console.print(summary)
 
-    chunks_table = Table(title="Chunks distribution")
-    chunks_table.add_column("Chunk type", justify="left", style="#00FFC8", no_wrap=True)
-    chunks_table.add_column("Size", justify="center", style="#00FFC8", no_wrap=True)
-    chunks_table.add_column("Ratio", justify="center", style="#00FFC8", no_wrap=True)
+        # fmt: off
+        chunks_table = Table(title="Chunks distribution")
+        chunks_table.add_column("Chunk type", justify="left", style="#00FFC8", no_wrap=True)
+        chunks_table.add_column("Size", justify="center", style="#00FFC8", no_wrap=True)
+        chunks_table.add_column("Ratio", justify="center", style="#00FFC8", no_wrap=True)
+        # fmt: on
 
-    for handler, size in sorted(
-        chunks_distribution.items(), key=lambda item: item[1], reverse=True
-    ):
-        chunks_table.add_row(
-            handler.upper(), human_size(size), f"{(size/total_size) * 100:0.2f}%"
-        )
+        for handler, size in sorted(
+            chunks_distribution.items(), key=lambda item: item[1], reverse=True
+        ):
+            chunks_table.add_row(
+                handler.upper(), human_size(size), f"{(size/total_size) * 100:0.2f}%"
+            )
 
-    console.print(chunks_table)
+        console.print(chunks_table)
```

Comment:
This whole logic change is unrelated to the display of output path. Should be placed in a separate commit.

Reply:
Separated. The logic change is a fix commit now, as in certain failure cases there was no user feedback due to an early return in the function.

Separated as fix-commit, moved `Chunks identification ratio` from `Summary` to after chunk distribution table, as chunk stat was causing the silent failure, this change lets both `Summary` and `Encountered errors` to be printed, while keeping chunk stat in one block.